src/string/src/lib.rs

```diff
@@ -12,13 +14,41 @@ use core::cmp;
 use core::usize;
 
 #[no_mangle]
-pub extern "C" fn memccpy(s1: *mut c_void, s2: *const c_void, c: c_int, n: usize) -> *mut c_void {
-    unimplemented!();
+pub unsafe extern "C" fn memccpy(
+    dest: *mut c_void,
+    src: *const c_void,
+    c: c_int,
+    n: usize,
+) -> *mut c_void {
+    use compiler_builtins::mem::memcpy;
+    use core::mem;
+    let dest = dest as *mut u8;
+    let to = memchr(src, c, n);
+    if to as usize == 0 {
```

Comment:
Raw pointers in Rust have the `is_null` method. `if to.is_null()` may be more readable.